### ISSUE TITLE
operator announces services that have no backend

### ISSUE THREAD

**Issue by ensonic** [authzed/spicedb-operator#113]
The code in https://github.com/authzed/spicedb-operator/blob/main/pkg/config/config.go#L520 unconditionally announces ports. If you are running with in memory datastore and hence no dispatch, that one should not be added (see PR#112).
Likewise the gateway 8443 should be conditional (some TLS settings).

**Comment by ensonic** [comment]
As a workaround I've tried:
```yaml
spec:
  config:
    httpEnabled: true
```
but that only sets:
```yaml
    env:                                                                                                                                      
    - name: SPICEDB_HTTP_ENABLED 
```
with no value and http is not enabled (aka gateway port is not actually open)

**Comment by ksingh7** [comment]
@ensonic  i have the same problem 


- Tried enabling HTTP (like below)
```
kubectl apply --server-side -f - <<EOF
apiVersion: authzed.com/v1alpha1
kind: SpiceDBCluster
metadata:
  name: dev
spec:
  config:
    image: ghcr.io/authzed/spicedb:latest
    datastoreEngine: memory
    httpEnabled: true
  secretName: dev-spicedb-config
---
apiVersion: v1
kind: Secret
metadata:
  name: dev-spicedb-config
stringData:
  preshared_key: "averysecretpresharedkey" 
EOF
```
- however in container logs , i don't see port 8443 listening , hence can't access application on REST API 8443
```
{"level":"info","format":"auto","log_level":"info","provider":"zerolog","async":false,"time":"2022-12-29T17:17:43Z","message":"configured logging"}
{"level":"info","v":0,"provider":"none","endpoint":"","service":"spicedb","insecure":false,"time":"2022-12-29T17:17:43Z","message":"configured opentelemetry tracing"}
{"level":"info","latest-released-version":"v1.15.0","time":"2022-12-29T17:17:44Z","message":"this is the latest released version of SpiceDB"}
{"level":"info","time":"2022-12-29T17:17:44Z","message":"using memory datastore engine"}
{"level":"warn","time":"2022-12-29T17:17:44Z","message":"in-memory datastore is not persistent and not feasible to run in a high availability fashion"}
{"level":"info","initialSlowRequest":"10ms","maxRequests":1000000,"hedgingQuantile":0.95,"time":"2022-12-29T17:17:44Z","message":"request hedging enabled"}
{"level":"info","maxCost":"16 MiB","numCounters":1000,"metrics":false,"time":"2022-12-29T17:17:44Z","message":"configured namespace cache"}
{"level":"info","maxCost":"564 MiB","numCounters":10000,"metrics":false,"time":"2022-12-29T17:17:44Z","message":"configured dispatch cache"}
{"level":"info","addr":":50051","network":"tcp","service":"grpc","workers":0,"insecure":true,"time":"2022-12-29T17:17:44Z","message":"grpc server started serving"}
{"level":"info","addr":":9090","service":"metrics","insecure":true,"time":"2022-12-29T17:17:44Z","message":"http server started serving"}
{"level":"info","addr":":8080","service":"dashboard","insecure":true,"time":"2022-12-29T17:17:44Z","message":"http server started serving"}
```

**Comment by ksingh7** [comment]
@vroldanbet as per your comment [1]

Looks like operator can't expose http/8443 , is there a workaround which i can try to enable 8443 and get it working for time being.

[1] https://github.com/authzed/spicedb-operator/pull/117#issuecomment-1346128515

**Comment by vroldanbet** [comment]
@ksingh7 so I think you have a different problem than the one described in this issue:
- this issue says that services announce ports even if there is no backend exposing it
- your problem is that you want to enable the HTTP Gateway, but the operator does not pick up the flag

would you mind opening another issue? As per workaround, I think it should be possible with `passthrough`. I can run some experiments, but let's move the conversation to a different issue 🙏🏻

**Comment by ecordell** [comment]
Fixed by #117
